src/main/java/hudson/plugins/analysis/core/PluginDescriptor.java

```diff
@@ -212,7 +203,12 @@ public FormValidation doCheckDefaultEncoding(@QueryParameter final String defaul
      */
     public FormValidation doCheckPattern(@AncestorInPath final AbstractProject<?, ?> project,
```

Comment:
Calls to this method only make sense for non-Workflow jobs (since WF jobs haven't a "traditional" workspace), so no need to use ```Job``` in the signature (below ```if (project != null)``` is taking care of doing it correctly).